### PR TITLE
Change weights downloading

### PR DESCRIPTION
**Pull request by kprokofi**

### CODE REVIEW

File: torchreid/models/model_store.py [hunk 1]
```diff
@@ -236,12 +237,17 @@ def load_model(net,
...
     from collections import OrderedDict
 
-    pretrained_dict = torch.load(file_path, map_location="cpu")
+    if file_path:
+        check_isfile(file_path)
+    pretrained_dict = (torch.load(file_path, map_location="cpu")
+                       if not pretrained_dict
+                       else pretrained_dict)
     model_dict = net.state_dict()
+    print(model_dict.keys())
```
Comment (sovrasov) [inline torchreid/models/model_store.py]: Delete

**Comment by kprokofi** [comment]
@sovrasov, are there any additional comments or merge? I want to update the submodule in OTE and run CI tonight

File: torchreid/utils/torchtools.py [hunk 2]
```diff
@@ -257,62 +258,3 @@ def count_num_param(model):
...
-        weight_path (str): path to pretrained weights.
-
-    Examples::
-        >>> from torchreid.utils import load_pretrained_weights
-        >>> weight_path = 'log/my_model/model-best.pth.tar'
-        >>> load_pretrained_weights(model, weight_path)
-    """
-    checkpoint = load_checkpoint(weight_path)
-    if 'classes_map' in checkpoint:
-        model.classification_classes = checkpoint['classes_map']
```
Comment (sovrasov) [inline torchreid/utils/torchtools.py:280]: This functionality is required for implementing classes filtering

File: tools/main.py [hunk 3]
```diff
@@ -50,7 +51,7 @@ def build_auxiliary_model(config_file, num_classes, use_gpu, device_ids=None, we
     model = torchreid.models.build_model(**model_kwargs(cfg, num_classes))
 
     if (weights is not None) and (check_isfile(weights)):
-        load_pretrained_weights(model, weights)
```
Comment (sovrasov) [inline tools/main.py]: `load_model` breaks --classes argument functionality, so I'd ask to keep `load_pretrained_weights` as is.
To solve the shape mismatch issue you could use ` load_pretrained_weights` inside `resume_from_checkpoint` to load the model.

File: tools/main.py [hunk 4]
```diff
@@ -137,9 +138,9 @@ def main():
     if cfg.model.load_weights and check_isfile(cfg.model.load_weights):
         if cfg.model.pretrained and not cfg.test.evaluate:
             state_dict = torch.load(cfg.model.load_weights)
-            model.load_pretrained_weights(state_dict)
+            load_model(model, pretrained_dict=state_dict)
```
Comment (sovrasov) [inline tools/main.py]: This if is seems to be useless in here. I propose to leave `load_pretrained_weights(model, cfg.model.load_weights)` from the second branch

File: tools/main.py (hunk 3 above)
Comment (kprokofi) [reply tools/main.py]: Got it. I will retain load_pretrained_weights instead of load_model then

**Review by sovrasov — APPROVED** [review]

**Pull request merged**